### PR TITLE
fix: Consistent agent display names across UI 

### PR DESCRIPTION
**Pull request by Akhileswara-Microsoft**
## Purpose
This pull request introduces a utility function to consistently format agent display names across the backend, ensuring that agent names are human-readable and match frontend conventions. The new formatting logic is applied throughout agent response handling and orchestration, replacing ad-hoc string manipulation with a centralized approach.

**Agent Display Name Formatting Improvements:**

* Added `format_agent_display_name` to `response_handlers.py`, converting raw agent IDs (like `HRHelperAgent` or `hr_helper_agent`) into human-readable names (like "HR Helper Agent"), mirroring frontend logic for consistency.
* Updated `agent_response_callback` and `streaming_agent_response_callback` to use `format_agent_display_name`, ensuring all agent names in responses and streaming updates are properly formatted. 

**Orchestration Consistency:**

* Modified `_process_event_stream` in `orchestration_manager.py` to use `format_agent_display_name` for agent headers and streaming messages, replacing previous underscore-to-space logic.

**Imports and Code Organization:**

* Added import of `format_agent_display_name` in `orchestration_manager.py` to support the new formatting logic.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## What to Check
This pull request introduces a utility function to standardize the formatting of agent display names across the backend, ensuring consistency with the frontend's naming conventions. The new function is integrated into several response handling and orchestration flows, so agent names are now presented in a more human-readable format throughout the application.

Key changes:

**Agent display name formatting:**

* Added a new `format_agent_display_name` function to `response_handlers.py` that converts raw agent identifiers (e.g., `HRHelperAgent`, `hr_helper_agent`) into human-readable display names (e.g., `HR Helper Agent`). This mirrors frontend logic for consistency.
* Updated `agent_response_callback` and `streaming_agent_response_callback` to use `format_agent_display_name` when setting agent names, ensuring all agent messages and tool calls display formatted names. 

**Orchestration integration:**

* Imported and used `format_agent_display_name` in `orchestration_manager.py` to format agent names in streaming event headers and message payloads, replacing previous ad-hoc formatting.

## Other Information

This pull request introduces a utility function to standardize the formatting of agent display names, ensuring consistency across backend and frontend components. The function is integrated into multiple places where agent names are shown to users, replacing previous ad-hoc formatting logic.

Key changes include:

**Agent Display Name Formatting:**

* Added a new `format_agent_display_name` function in `src/backend/callbacks/response_handlers.py` to convert raw agent IDs (e.g., `hr_helper_agent`, `HRHelperAgent`) into human-readable display names (e.g., "HR Helper Agent"), mirroring frontend logic for consistent naming.

**Integration of Formatting Utility:**

* Updated `agent_response_callback` and `streaming_agent_response_callback` in `src/backend/callbacks/response_handlers.py` to use `format_agent_display_name` when rendering agent names, ensuring all agent messages display standardized names. 

* Modified `_process_event_stream` in `src/backend/orchestration/orchestration_manager.py` to use `format_agent_display_name` for agent section headers and streaming messages, replacing previous underscore-based formatting.

* Updated imports in `src/backend/orchestration/orchestration_manager.py` to include the new formatting function.

### CODE REVIEW

File: src/backend/callbacks/response_handlers.py [hunk 1]
```diff
@@ -17,6 +17,44 @@
...
+        return "Assistant"
+
+    name = raw_name
+
+    # Replace underscores with spaces
+    name = name.replace("_", " ")
+
+    # Insert space before each uppercase letter preceded by a lowercase letter
+    # e.g. "HRHelperAgent" → "H R Helper Agent" (handled next)
+    name = re.sub(r'([a-z])([A-Z])', r'\1 \2', name)
```
[the remark this answers is not in the document]
Comment (Akhileswara-Microsoft) [reply src/backend/callbacks/response_handlers.py:38]: uodate comment

**Review by Akhileswara-Microsoft — COMMENTED** [review]
All 5 Copilot review comments addressed in commit 2be207c0:

1. **str.replace substring bug** - Replaced with word-boundary regex \
e.sub(r'\\b...\\b', ...)\. Added test verifying \HabitAgent\ -> \Habit Agent\ (not \HabIT Agent\).
2. **Inaccurate comment** - Updated to \HelperAgent -> Helper Agent\ matching actual regex behavior.
3. **Missing unit tests** - Added 11 tests in \TestFormatAgentDisplayName\ (empty/None, PascalCase, snake_case, acronyms, word-boundary safety, etc.).
4. **Breaking existing tests** - Updated all assertions: \TestAgent\ -> \Test Agent\, \gent_123\ -> \Agent 123\. 52 tests pass.
5. **Docstring mismatch** - Updated to clarify function does NOT strip 'Agent' suffix (frontend handles that separately).

File: src/backend/callbacks/response_handlers.py [hunk 2]
```diff
@@ -66,6 +104,7 @@ def agent_response_callback(
     Final (non-streaming) agent response callback using agent_framework Message.
     """
     agent_name = getattr(message, "author_name", None) or agent_id or "Unknown Agent"
+    agent_name = format_agent_display_name(agent_name)
     role = getattr(message, "role", "assistant")
```
[the remark this answers is not in the document]
Comment (Akhileswara-Microsoft) [reply src/backend/callbacks/response_handlers.py:109]: done testcase update.

**Review by Avijit-Microsoft — APPROVED** [review]
